concrete/src/Logging/HttpLoggableAwareTrait.php

```diff
@@ -0,0 +1,155 @@
+
+/**
+ * Class HttpLoggableAwareTrait
+ *
+ * Check if a http request or http response body can be logged (doesn't contain binary data)
+ */
+trait HttpLoggableAwareTrait
+{
+
+    /**
+     * Check if request body can be logged (is not binary)
+     *
+     * @param ZendRequest|SymfonyRequest|RequestInterface $request
+     * @return bool
+     */
+    public function isRequestBodyLoggable($request)
+    {
+        if ($request instanceof SymfonyRequest) {
+            $headers = array_change_key_case($request->headers->all());
+        } elseif ($request instanceof ZendRequest) {
+            $headers = array_change_key_case($request->getHeaders()->toArray());
+        } elseif ($request instanceof RequestInterface) {
+            $headers = array_change_key_case($request->getHeaders());
+        } else {
+            $headers = [];
+        }
+
+        return $this->isLoggable($headers, $this->getLoggableContentTypes());
+    }
+
+    /**
+     * Check if response body can be logged (is not binary)
+     *
+     * @param ZendResponse|SymfonyResponse|ResponseInterface $response
+     * @return bool
+     */
+    public function isResponseBodyLoggable($response)
+    {
+        if ($response instanceof SymfonyResponse) {
+            $headers =  array_change_key_case($response->headers->all());
+        } elseif ($response instanceof ZendResponse) {
+            $headers =  array_change_key_case($response->getHeaders()->toArray());
+        } elseif ($response instanceof ResponseInterface) {
+            $headers = array_change_key_case($response->getHeaders());
+        } else {
+            $headers = [];
+        }
+
+        return $this->isLoggable($headers, $this->getLoggableContentTypes());
+    }
+
+    /**
+     * Compare the headers of the request or response with the whitelist
+     * If the content type matches a entry in the whitelist,
+     * the response or request body can be logged
+     *
+     * @param array $headers
+     * @param array $types
+     * @return bool
+     */
+    public function isLoggable(array $headers, array $types)
+    {
+        $isLoggable = false;
+
+        if (array_key_exists('content-type', $headers)) {
+            $responseContentType = $headers['content-type'];
+
+            if (is_array($types)) {
+                foreach($types as $type) {
+                    $type = str_replace( '*' , '(.*)', $type);
+                    $search = '#^'.$type.'#i'  ;
+                    preg_match($search, $responseContentType, $matches);
+
+                    if (count($matches)) {
+                        $isLoggable = true;
+                        break;
+                    }
+                }
+            }
+        }
+        return $isLoggable;
+    }
+
+    /**
+     * Get whitelist of http content types which can be logged
+     *
+     * @return array
+     */
+    public function getLoggableContentTypes()
+    {
+        $config = \Core::make('config');
```

Comment:
I don't like making the `Http\Client` application aware. So I used `\Core::make()`. Maybe that's wrong.Any better solutions?

Reply:
You can also use `$config = app('config');`